pkg/scheduler/scheduler.go

```diff
@@ -599,10 +608,10 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 			// Pod did not fit anywhere, so it is counted as a failure. If preemption
 			// succeeds, the pod should get counted as a success the next time we try to
 			// schedule it. (hopefully)
-			metrics.PodScheduleFailures.Inc()
+			sched.podScheduleFailureCounter.Inc()
```

Comment:
Will it be better to call metrics. PodScheduleFailures() here instead of storing the counter in the scheduler?  Maybe we should keep scheduler and metrics decoupling?

Reply:
`PodScheduleFailures()` call causes `(*lazyMetric) IsCreated()` method and it get read lock([code](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/metrics/metric.go#L76-L80)).  So, calling `PodScheduleFailures()` each time increases overhead.  I think its trade-off.  Should we take decoupling or performance?

Reply:
Then how about just delay the initialize of scheduler/metrics.{PodScheduleSuccesses, PodScheduleFailures, PodScheduleErrors}. It can be moved to metrics.Register(), I guess.

1. Declare them and initialize them with `noop`.
2. Re-assign them to xxx.With() in metrics.Register().

WDYT

Reply:
ah, nice idea.  I'll do so.  Thanks for the idea.